compiler/shared_ast/expr.ml

```diff
@@ -764,10 +813,6 @@ let make_default_unboxed excepts just cons =
     | _ -> None
   in
   match excepts, bool_value just, cons with
-  | [], Some true, cons -> Marked.unmark cons
```

Comment:
We could probably keep at least the 2nd and 3rd cases here ; the comment in the .mli stating that the result may not be of the form `EDefault` should be updated since we now ensure that's the case.

Also, I have a doubt about it but I remember this being tied with the "cumulative exceptions" features, which might be a bit fragile, so it'd be worth double-checking that disabling this doesn't break it.

Reply:
I let you check that cumulative exceptions still work. I reestablished the second and third optimizations.